tests/unit_tests/Test/PM/Unit/Mappings.cls

```diff
@@ -0,0 +1,463 @@
+/// Unit tests for CLS, PKG, LOC, MAC mappings
+Class Test.PM.Unit.Mappings Extends %UnitTest.TestCase
+{
+
+Property Params As %Library.DynamicObject;
+
+Property ResourceReferenceClass As %IPM.Storage.ResourceReference;
+
+Property Default As %IPM.Storage.ModuleSetting.ProcessorDefault;
+
+Method OnBeforeAllTests() As %Status
+{
+    Set ..Params = ##class(%Library.DynamicObject).%New()
+    Do ..Params.%Set("Verbose", 0)
+    Do ..Params.%Set("DeveloperMode", 0)
+    Do ..Params.%Set("NoMapping", 0)
+    Do ..Params.%Set("Compile", 1)
+    Do ..Params.%Set("Phase", "Reload")
+    Do ..Params.%Set("Namespace", "USER")
+    Do ..Params.%Set("ResourceName", "")
+    Do ..Params.%Set("ModuleName", "DefaultModule")
+
+    Set ..ResourceReferenceClass = ##class(%IPM.Storage.ResourceReference).%New()
+    Set ..ResourceReferenceClass.Module = ##class(%IPM.Storage.Module).%New()
+    Set ..ResourceReferenceClass.Module.Defaults = ##class(%ListOfObjects).%New()
+    Set ..Default = ##class(%IPM.Storage.ModuleSetting.ProcessorDefault).%New()
+    Set ..Default.Conditions = ##class(%ListOfObjects).%New()
+    Quit $$$OK
+}
+
+Method TestNoParentMappingsMAC()
+{
+    // Initialize ..ResourceReference as MAC file
+    Set ..ResourceReferenceClass.Name = "Foo.Bar.MAC"
+
+    // Create a default processor setting for MAC
+    Set ..Default.Class = "%IPM.ResourceProcessor.Default.Routine"
+
+    // Add the default setting to the module
+    Do ..ResourceReferenceClass.Module.Defaults.Insert(..Default)
+
+    Set testObjMAC = ##class(%IPM.ResourceProcessor.Default.Routine).%New(..ResourceReferenceClass)
+    Set ns = "USER"
+    Set parentRoutine = "Foo"
+    Set childRoutine = "Foo.Bar"
+    Set db = "HSLIB"
```

Comment:
@isc-jili Looks good but HSLIB doesn't exist on vanilla IRIS so would throw an error. You can create your own database at the beginning of the test and delete it by the end (OnBefore/AfterAllTests) or could map from say IRISSYS/another db that always exists?